### ISSUE TITLE
Egsam test doesn't timeout

### ISSUE THREAD

**Issue by makeworld-the-better-one** [makeworld-the-better-one/go-gemini#6]
Test number three on page `gemini://egsam.pitr.ca/1.1.gmi` still fails and will wait for over 15 seconds.

Go [code](https://github.com/pitr/egsam/blob/ff6cc266f10d4a4f51c6c329c716884fae080521/main.go#L17) is:
```go
	g.Handle("/1.1.write.timeout", func(c gig.Context) error {
		time.Sleep(10 * time.Minute)
		return c.Gemini("FAILED. You waited too long")
	})
```

There needs to be a timeout setting in the client code that will cause a timeout and return if **the connection is opened, but no response is given**.

**Comment by makeworld-the-better-one** [comment]
`conn.SetDeadline` should be able to be used. Can the other timeout method be removed once that is added? Also, there needs to be a way to detect a timeout error through the API.

**Comment by makeworld-the-better-one** [comment]
This has been fixed over several commits: 2015958ad2af0008b64b83be41382405be8260f4...d842c9696bc2d4d2dec719253df73d7fecf79c70

Using `Client.ConnectTimeout` one can set a timeout that applies to forming a connection as well as receiving a header. As long as this value is set (and it is by `DefaultClient`) then all the 1.1 Egsam tests are fully passed.
